components/shared/script/lib.rs

```diff
@@ -1327,3 +1330,38 @@ impl SerializedImageData {
         }
     }
 }
+
+#[derive(
+    Clone, Copy, Debug, Deserialize, Eq, Hash, MallocSizeOf, Ord, PartialEq, PartialOrd, Serialize,
+)]
+/// Index of gamepad in list of system's connected gamepads
+pub struct GamepadIndex(pub usize);
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+/// The minimum and maximum values that can be reported for axis or button input from this gamepad
+pub struct GamepadInputBounds {
+    /// Minimum and maximum axis values
+    pub axis_bounds: (f64, f64),
+    /// Minimum and maximum button values
+    pub button_bounds: (f64, f64),
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+/// The type of Gamepad event
+pub enum GamepadEvent {
+    /// A new gamepad has been connected
+    Connected(GamepadIndex, String, GamepadInputBounds),
+    /// An existing gamepad has been disconnected
+    Disconnected(GamepadIndex),
```

Comment:
https://w3c.github.io/gamepad/#event-gamepaddisconnected

(It never hurst to add links, even if they are also added at the point of event-handling already)